### ISSUE TITLE
EU.ORG got blocked by GFW recently

### ISSUE THREAD

**Issue by xtexChooser** [net4people/bbs#350]
About one week ago, some people reported that TLS connections to EU.ORG domains are blocked by the Great Firewall.
Can be confirmed in both China Telecom and China Mobile networks.

There are no known DNS pollution in this blocking. All DNS queries I tested got correct result.

Plain HTTP requests on 80/TCP are not blocked.
After attempts to establish a TLS connection on 443/TCP (other ports are not tested), the connection will be reseted, and further packets to the server's 443/TCP will be dropped for several minutes.

**Comment by systemhorse** [comment]
Note: This is a free domain name provider. It has no relation with the European Union, despite its name.

**Comment by gaukas** [comment]
Can you confirm is this a blocking against the SNI `eu.org` or against part/all of IP addresses EU.ORG resolves to?

**Comment by xtexChooser** [comment]
@gaukas It is against the SNI. I tested:
1. DNS query a domain not under `eu.org`
2. Connect, HTTPS connections to that domain are not blocked
3. DNS query a domain does under `eu.org` which resolves to the same address as the first one
4. TLS connect to the first domain, not blocked
5. Attempt to establish a TLS connection with the second domain, getting blocked
6. Connections to the first domain are blocked

**Comment by gaukas** [comment]
Thanks! So after a TLS handshake using blocked SNI with a target, all TLS connections (supposedly from the same source IP) to that server (IP:443) are blocked for some period of time.

**Comment by gaukas** [comment]
> this happened before

So what is the significance of eu.org then 🧐

> blocked by SNI or by preceding DNS query 

Step 3, 4, 5 supported that it is due to SNI as I can see.

**Comment by xtexChooser** [comment]
> I don't believe all free domain tlds are targeted?

note that eu.org is not a TLD.

**Comment by gaukas** [comment]
> not a TLD.

That's true, it is my bad for not stating clearly my question: since there are plenty of choices for free tlds and other free subdomains, there is no reason to target `eu.org` unless it is special in some ways.

**Comment by def24** [comment]
Just one of your "subdomains" being targetted is enough they would targert *.maindomain. We have seen this hundreds of times. So it does not need to be special.

**Comment by gaukas** [comment]
> We have seen this hundreds of times.

Thank you for sharing. I'm not aware of this, could you please point me to other discussing threads or other resources about the same behavior? 

And I wonder if this implies none of the free subdomains will be available in China, perhaps also including restrictive ones such as *.netlify.app, *.azurewebsites.net, etc?

**Comment by gaukas** [comment]
I just learned that this blocking behavior has been lifted on `*.eu.org`, can anyone confirm?

**Comment by xtexChooser** [comment]
yes seems for me.

**Comment by wkrp** [comment]
Reopen to keep discussion visible.
